extra_data/reader.py

```diff
@@ -1617,6 +1618,13 @@ def __getitem__(self, aliased_item):
 
         raise TypeError('expected alias or (source alias, key) tuple')
 
+    def __contains__(self, aliased_item):
+        try:
+            self[aliased_item]
+            return True
+        except:
```

Comment:
EDIT: Saw your fix-up in the meantime. Please also test raising the exception.

Reply:
I think this is already in ec51e06? Valid/invalid aliases are tested.

Reply:
Doh. Of course you're catching the exception in the first place, there's nothing to test whether it raises one. Moving right long! (I would still prefer `KeyError` over `Exception` though!)

Reply:
My reasoning was to catch `Exception` because `__getitem__()` can throw a `TypeError` as well :thinking: But maybe it does make sense to throw a `TypeError` anyway?

Reply:
(or rather, not catch the `TypeError`)

Reply:
I would argue causing a `TypeError` in `__getitem__` should propagate to indicate your alias value wasn't valid in the first place rather than not contained in the collection. This follows what built-in types seem to do, e.g.:
```python
>>> () in set()
False
>>> [] in set()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'list'
```

Reply:
Makes sense, fixed in 46a8197.